### DISCUSSION TITLE
Running into problems during DataflowPartitioning

### DISCUSSION THREAD

**Discussion by Adityasrinivas24** [Xilinx/finn#1051] [Q&A]
Hi, I'm following the cnv_end2end example for my custom defined Neural Network. I streamlined the model as shown in the example and tried using `CreateDataflowPartition()` but ran into `AssertionError: cycle-free graph violated: partition depends on itself`. 

I went through a couple of older issues  but they didn't fix my problem. Is there a standard flow that i can use to perform the transformations? How do i figure out problems in netron or other utility? 
Ive also attached the network visualised in netron after streamlining. Any help will be appreciated .

Thank you
![ecgnet1_w8a8_streamlined onnx](https://github.com/Xilinx/finn/assets/106761063/63c9e343-44d4-415c-852f-e194c91125c5)

**Comment by Arbiter-glitch** [comment]
I have never dealt with a model with Maxpool. However, the transpose nodes should be absorbed into the multithreshold. Also, the repeated Multithreshold nodes can be collapsed into a single node. Check this [link](https://finn.readthedocs.io/en/latest/source_code/finn.transformation.streamline.html) and refer the transformations there it might help you.

**Reply by Adityasrinivas24** [reply]
Thank you for the reply. I'll look into the docs again. I think the Maxpool layers are causing the partitioning problem. I'll try to find alternatives. 
Thanks again
